File: exercises/week02/part03/cc/src/trainee/trainee.cc
#include "trainee.h"

Trainee::Trainee(void) {
  printf("Trainee class constructor.\n");
}

Trainee::Trainee(const std::string& name, const std::string& student_id, const std::string& job, const float& salary)
  : Student(name, student_id), _job(job), _salary(salary) {
    printf("Trainee class parameterized constructor.\n");
  }

Trainee::~Trainee() {
  printf("Trainee %s destructed.\n", this->name().c_str());
}

//getters

std::string Trainee::job(void) const { return _job; } 
float Trainee::salary(void) const { return _salary; }

//setters

void Trainee::set_job(const std::string& job) {
  _job = job;
}

void Trainee::set_salary(const float& salary) {
  _salary = salary;
}

//public methods

void Trainee::print(void) const {
  Student::print();
  printf("Job: %s\nSalary: %.2f\n", _job.c_str(), _salary);
}

File: exercises/week02/part02/Trainee.java
package part02;

public class Trainee extends Student {
  private String job;
  private float salary;

  public Trainee() {}

  public Trainee(String name, String studentId, String job, float salary) {
    this.setName(name);
    this.setStudentId(studentId);
    this.job = job;
    this.salary = salary;
  }

  //getters

  public String getJob() { return this.job; }
  public float getSalary() { return this.salary; }

  //setters

  public void setJob(String job) {
    this.job = job;
  }

  public void setSalary(float salary) {
    this.salary = salary;
  }

  //public methods

  public void print() {
    super.print();
    var info = String.format("Job: %s\nSalary: %.2f",
      this.job,
      this.salary
    );
    System.out.println(info);
  }
}
